samples/custom_dispatch/cpu/plugin/mlp_plugin.c

```diff
@@ -0,0 +1,212 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Demonstrates an mlp example with the implementation of MLP provided using system linked plugin exporting a single `mlp_external` function.
+// See samples/custom_dispatch/cpu/plugin/system_plugin.c for more information about system plugins and their caveats.
+
+#include <inttypes.h>
+#include <stdio.h>
+
+// The only header required from IREE:
+#include "iree/hal/local/executable_plugin.h"
+
+// Stateful plugin instance.
+// There may be multiple of these in a process at a time, each with its own
+// load/unload pairing. We pass a pointer to this to all import calls via the
+// context argument.
+typedef struct {
+  iree_hal_executable_plugin_allocator_t host_allocator;
+  FILE* file;
```

Comment:
you can drop `file` and all the stuff mentioning it, as it's not relevant to this sample

Reply:
I kept it cause that way I can check that I actually invoked the code within the plugin.